homeassistant/components/amberelectric/sensor.py

```diff
@@ -0,0 +1,591 @@
+"""Amber Electric Sensor definitions."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+from typing import Any
+
+import amberelectric
+from amberelectric.api import amber_api
+from amberelectric.model.channel import ChannelType
+from amberelectric.model.interval import SpikeStatus
+
+from homeassistant.components.sensor import DEVICE_CLASS_MONETARY, SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_ATTRIBUTION
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+from homeassistant.util import slugify
+
+from .const import CONF_API_TOKEN, CONF_SITE_ID, LOGGER
+from .coordinator import AmberDataService
+
+ATTRIBUTION = "Data provided by Amber Electric"
+
+
+def friendly_channel_type(channel_type: str) -> str:
+    """Return a human readable version of the channel type."""
+    if channel_type == ChannelType.CONTROLLED_LOAD:
+        return "Controlled Load"
+    if channel_type == ChannelType.FEED_IN:
+        return "Feed In"
+    return "General"
+
+
+class AmberPriceSensor(CoordinatorEntity, SensorEntity):
+    """Amber Price Sensor."""
+
+    def __init__(
+        self,
+        platform_name: str,
+        site_id: str,
+        channel_type: str,
+        data_service: AmberDataService,
+        coordinator: DataUpdateCoordinator,
+    ) -> None:
+        """Initialize the Sensor."""
+        super().__init__(coordinator)
+        self._site_id = site_id
+        self._channel_type = channel_type
+        self._platform_name = platform_name
+        self._data_service = data_service
+
+    @property
+    def name(self) -> str | None:
+        """Return the friendly name of the sensor."""
+        return (
+            self._platform_name
+            + " - "
+            + friendly_channel_type(self._channel_type)
+            + " "
+            + "Price"
+        )
+
+    @property
+    def unique_id(self) -> str | None:
+        """Return a unique id for each sensors."""
+        return slugify(
```

Comment:
why slugify ?

Reply:
I didn't realise the id logic kind of did this already. Refactored now.